packages/odata-common/src/request/odata-request.ts

```diff
@@ -135,17 +143,21 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
    * @returns Query parameter string
    */
   query(): string {
-    const query = Object.entries(this.queryParameters())
-      .map(([key, value]) => `${key}=${value}`)
-      .join('&');
-    return query.length ? `?${query}` : '';
+    const parameters = getOptionWithPriority(this.queryParameters());
+    if (parameters) {
+      const query = Object.entries(parameters)
+        .map(([key, value]) => `${key}=${value}`)
+        .join('&');
+      return query.length ? `?${query}` : '';
+    }
+    return '';
   }
 
   /**
    * Create object containing all headers, including custom headers for the given request.
    * @returns Key-value pairs where the key is the name of a header property and the value is the respective value
    */
-  async headers(): Promise<Record<string, any>> {
+  async headers(): Promise<Record<string, ValueWithOrigin>> {
```

Comment:
I would prefer to a return a list of KeyValueWithOrigin or if you like to keep the OptionsFromOrigin a list of these so that no merging takes place until the http-client does the execute.